cli/azd/pkg/project/service_config.go

```diff
@@ -58,6 +58,9 @@ type ServiceConfig struct {
 	// Environment variables to set for the service
 	Environment osutil.ExpandableMap `yaml:"env,omitempty"`
 
+	// AdditionalProperties captures any unknown YAML fields for extension support
+	AdditionalProperties map[string]interface{} `yaml:",inline"`
```

Comment:
It feels a little weird having both `Config` and `AdditionalProperties` - should extensions like the agents extension just use `AdditionalProperties` going forward?

Reply:
I'd like to keep additional properties since they do solve different problems... ServiceConfig.Config is loosely typed but often typed and used by extensions.  Example - custom service target config options

Versus additional properties - Loosely typed property bag that can be used by extensions to set cross cutting config concerns across multiple services or on the project itself.